api/repositories/service_binding_repository_test.go

```diff
@@ -1238,4 +1239,114 @@ var _ = Describe("ServiceBindingRepo", func() {
 			})
 		})
 	})
+
+	Describe("GetServiceBindingDetails", func() {
+		var (
+			serviceBindingGUID   string
+			getErr               error
+			cfServiceInstance    *korifiv1alpha1.CFServiceInstance
+			serviceBindingRecord repositories.ServiceBindingDetailsRecord
+			credentialsSecret    *corev1.Secret
+		)
+
+		BeforeEach(func() {
+			serviceInstanceGUID := prefixedGUID("instance")
+			cfServiceInstance = createServiceInstanceCR(ctx, k8sClient, serviceInstanceGUID, space.Name, "service-instance-name", serviceInstanceGUID)
+			serviceBindingGUID = prefixedGUID("binding")
+
+			credentials := map[string]any{
+				"foo": "val1",
+				"bar": "val2",
+			}
+
+			credentialBytes, err := json.Marshal(credentials)
+			Expect(err).To(BeNil())
+
+			creds := map[string][]byte{
+				"credentials": credentialBytes,
+			}
+
+			credentialsSecret = &corev1.Secret{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      cfServiceInstance.Spec.SecretName,
+					Namespace: space.Name,
+					Labels: map[string]string{
+						"korifi.cloudfoundry.org/service-instance-guid": cfServiceInstance.Name,
+					},
+				},
+				Data: creds,
+			}
+			Expect(
+				k8sClient.Create(ctx, credentialsSecret),
+			).To(Succeed())
+
+			sbDisplayName := "test-service-binding"
+			serviceBinding := &korifiv1alpha1.CFServiceBinding{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      serviceBindingGUID,
+					Namespace: space.Name,
+					Labels:    map[string]string{"servicebinding.io/provisioned-service": "true"},
+				},
+				Spec: korifiv1alpha1.CFServiceBindingSpec{
+					DisplayName: &sbDisplayName,
+					Service: corev1.ObjectReference{
+						Kind:       "CFServiceInstance",
+						Name:       cfServiceInstance.Name,
+						APIVersion: "korifi.cloudfoundry.org/v1alpha1",
+					},
+					AppRef: corev1.LocalObjectReference{
+						Name: appGUID,
+					},
+					Type: "app",
+				},
+			}
+
+			Expect(
+				k8sClient.Create(ctx, serviceBinding),
+			).To(Succeed())
+
+			Expect(
+				k8s.Patch(ctx, k8sClient, serviceBinding, func() {
+					serviceBinding.Status.Credentials.Name = serviceInstanceGUID
```

Comment:
`serviceBinding.Status.Credentials.Name` is set to the secret that contains the credentials (see [here](https://github.com/cloudfoundry/korifi/blob/a3f8b883b33bf18e166a8d9385c134439e9ec3b9/controllers/controllers/services/bindings/managed/controller.go#L220)), it has nothing to do with the service instance. 
Therefore, the test could just create a credentials secret and set its name on the binding status, no need to create a service instance